### ISSUE TITLE
Layers Control has a WCAG Level A Empty Link Error

### ISSUE THREAD

**Issue by Cyngael13** [Leaflet/Leaflet#6821]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
Open a leaflet map with a layers control that is collapsed by default, e.g. https://leafletjs.com/examples/layers-control/example.html
Check it with a WCAG checker such as WebAim's WAVE tool: http://wave.webaim.org/report#/https://leafletjs.com/examples/layers-control/example.html

**Expected behavior**
The Layer Controls button needs text or context. Success Criterion 2.4.4 Link Purpose (In Context): https://www.w3.org/TR/WCAG21/#link-purpose-in-context

**Current behavior**
The hamburger button has no context on the link to expand the Layer Controls, and throws an error when people are surface-level checking for accessibility issues.

**Environment**
- Leaflet version: All
- Browser (with version): All
- OS/Platform (with version): All

**Solutions**
This can easily be solved with an aria-label.  https://www.w3.org/WAI/GL/wiki/Using_aria-label_for_link_purpose
```
<div class="leaflet-top leaflet-right">
<div class="leaflet-control-layers leaflet-control" aria-haspopup="true">
<a class="leaflet-control-layers-toggle" href="#" title="Layers"></a>
<section class="leaflet-control-layers-list"><div class="leaflet-control-layers-base">
```

to something like:
```
<div class="leaflet-top leaflet-right">
<div class="leaflet-control-layers leaflet-control" aria-haspopup="true">
<a class="leaflet-control-layers-toggle" href="#" title="Layers" aria-label="Toggle Layer Contol"></a>
<section class="leaflet-control-layers-list"><div class="leaflet-control-layers-base">
```

I just don't know how to do that in GitHub yet...

**Comment by Malvoz** [comment]
> **Expected behavior**
The Layer Controls button needs text or context. Success Criterion 2.4.4 Link Purpose (In Context): https://www.w3.org/TR/WCAG21/#link-purpose-in-context

It should be noted that the layer control is intended as a _button_ (although it is [currently missing `role="button"`](https://github.com/Leaflet/Leaflet/issues/7498)), and <q>SC 2.4.4 Link Purpose (In Context)</q> is only applicable to _links_. Though I'm not opposed to labelling the layer control using `aria-label`.
